### PR TITLE
Upgrade QISKit depdencies

### PR DESCRIPTION
**Pull request by atilag**
From time to time, it's good to keep our external dependencies upgraded with latest stable versions.
We have updated most of them, so many reported issues related to different environment setups can be fixed.
Thanks @afu1 for your fixes!!

## How Has This Been Tested?
make test (locally)
make lint (locally)
I had to fix some tests though.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: qiskit/dagcircuit/_dagcircuit.py [hunk 1]
```diff
@@ -640,8 +639,7 @@ def compose_front(self, input_circuit, wire_map=None):
         # Compose
         self.basis = union_basis
         self.gates = union_gates
-        ts = nx.topological_sort(input_circuit.multi_graph, reverse=True)
-        for n in ts:
+        for n in nx.topological_sort(input_circuit.multi_graph, reverse=True):
```
Comment (atilag) [inline qiskit/dagcircuit/_dagcircuit.py]: @afu1 the `reverse` parameter has been removed from the latest version of NetworkX. I was going to fix this by removing this parameter. Do you happen to know if we really need the reversed circuit here? ... If it's needed or we are not sure, we can keep the same logic with:
```python
list(reversed(list(nx.topological_sort(input_circuit.multi_graph))))
```

Comment (afu1) [reply qiskit/dagcircuit/_dagcircuit.py]: I quickly checked. This does not solve the issue with the test but I would use what you propose above for now. I don't think you need the outermost list though. If we bump the version to networkx 2.0 then I will go through the code again anyway and update some functions that are only there for compatibility with 1.11.

Comment (afu1) [reply qiskit/dagcircuit/_dagcircuit.py]: I ran make test on my modification and with the output of the random circuit I verified that the result was the same except for the statistical error in the counts. And I also ran many circuits (5,16,20 qubits) and the mapper worked without errors.

**Comment by diego-plan9** [comment]
I have added a commit for fixing a typo and revising the minimal version of the dependencies, and it seems one test is failing in Appveyor Python 3.5 - the most likely reason is that it seems to be using an outdated version of miniconda (4.2.12) https://www.appveyor.com/docs/build-environment/#miniconda, which in turn causes it to use the older version of networkx.

**Pull request merged**